crypto/lhash/lhash.c

```diff
@@ -186,16 +186,35 @@ void OPENSSL_LH_doall_arg(OPENSSL_LHASH *lh, OPENSSL_LH_DOALL_FUNCARG func, void
 static int expand(OPENSSL_LHASH *lh)
 {
     OPENSSL_LH_NODE **n, **n1, **n2, *np;
-    unsigned int p, i, j;
-    unsigned long hash, nni;
+    unsigned int p, pmax, nni, i, j;
+    unsigned long hash;
+
+    nni = lh->num_alloc_nodes;
+    p = lh->p;
+    pmax = lh->pmax;
+    if (p + 1 >= pmax) {
+        j = nni * 2;
+        n = OPENSSL_realloc(lh->b, sizeof(OPENSSL_LH_NODE *) * j);
+        if (n == NULL) {
+            lh->error++;
+            return 0;
+        }
+        for (i = nni; i < j; i++)
```

Comment:
Make this a single memset call.  Then you can get rid of either nni or j variables.

Reply:
@richsalz I see we're continuing to assume/require that NULL is zero-filled :-)

Reply:
Yes.  And in fact we test for it, see test/sanitytest.c